pkg/tenancy/tenancy.go

```diff
@@ -45,7 +45,10 @@ func GetTenantFromHTTP(r *http.Request, tenantHeader string, defaultTenantID str
 	var err error
 	tenant := r.Header.Get(tenantHeader)
 	if tenant == "" {
-		tenant = defaultTenantID
+		tenant = r.Header.Get(DefaultTenantHeader)
+		if tenant == "" {
+			tenant = defaultTenantID
+		}
```

Comment:
do you think its worth logging down below if we end up overwriting based on the cert? or maybe we could just handle this on startup with a log or an error?

Reply:
It might be too spammy. We had similar logs when parsing the tenant information from the request in the Store GW and they were removed: https://github.com/thanos-io/thanos/commit/510c05448d4f1af9e04c74b72e54559c49c785b9